File: code/community/1304606576094482442/opt-out-model-improvement-dotnet-sdk.js
// JavaScript Node.js example using axios
// Ensure 'axios' library is installed: `npm install axios`

const axios = require('axios');
require('dotenv').config();

const apiKey = process.env.DEEPGRAM_API_KEY;
const audioUrl = process.env.AUDIO_FILE_URL;

const url = 'https://api.deepgram.com/v1/listen?mip_opt_out=true';

axios.post(url, {
    url: audioUrl
}, {
    headers: {
        'Authorization': `Token ${apiKey}`,
        'Content-Type': 'application/json'
    }
})
.then(response => {
    console.log('Transcription:', response.data);
})
.catch(error => {
    console.error('Failed to transcribe audio:', error.response.status);
});

File: code/community/1304606576094482442/opt-out-model-improvement-dotnet-sdk.py
# Python example using requests
# Ensure 'requests' library is installed: `pip install requests`

import os
import requests

# Load environment variables
api_key = os.getenv('DEEPGRAM_API_KEY')
audio_url = os.getenv('AUDIO_FILE_URL')

url = 'https://api.deepgram.com/v1/listen?mip_opt_out=true'

headers = {
    'Authorization': f'Token {api_key}',
    'Content-Type': 'application/json',
}

data = {
    "url": audio_url
}

response = requests.post(url, headers=headers, json=data)

if response.ok:
    print("Transcription:", response.json())
else:
    print("Failed to transcribe audio:", response.status_code)